scripts/environment/gsutil_env.py

```diff
@@ -0,0 +1,281 @@
+
+        while True:
+            bucket_choice = raw_input(
+                ('{}\n'
+                 '[1] Enter bucket name\n'
+                 '[2] Choose from list\n'
+                 'Enter your numeric choice: '
+                 .format(prompt_header))).strip()
+
+            bucket = None
+            if bucket_choice == '1':
+                bucket = self._create_or_reuse_bucket(object_type)
+            elif bucket_choice == '2':
+                bucket = self._choose_bucket(output)
+            else:
+                continue
+
+            if bucket:
+                if not bucket.endswith('/'):
+                    bucket = bucket + '/'
+
+                if object_type == ScannerBucketObject.POLICY:
+                    # if policy bucket, enable versioning
+                    self.policy_bucket = bucket
+                    self._enable_object_versioning(bucket)
+                elif object_type == ScannerBucketObject.OUTPUT:
+                    # warn if using same buckets, prefer not to version
+                    # output and temp files
+                    if bucket == self.policy_bucket:
+                        same_bucket = raw_input(
+                            'I don\'t recommend using the same bucket for '
+                            'both your known-good policies AND output files!\n'
+                            'Continue anyway? y/N ').strip().lower()
+                        if same_bucket != 'y':
+                            continue
+                    self.output_bucket = bucket
+
+                print 'Using bucket: {}\n'.format(bucket)
+                break
+
+    def _create_or_reuse_bucket(self, object_type):
+        """
+        Give the user the option to either create or re-use a bucket.
+
+        This method can be used for either the POLICY or the OUTPUT
+        bucket types.
+        """
+        bucket_name = None
+
+        default_bucket = ''
+        if object_type == ScannerBucketObject.POLICY:
+            default_bucket = self.DEFAULT_BUCKETNAME_POLICIES
+        elif object_type == ScannerBucketObject.OUTPUT:
+            default_bucket = self.DEFAULT_BUCKETNAME_OUTPUT
+
+        default_bucket = default_bucket.format(self.gcloud_config.project_id)
+
+        while True:
+          bucket_name = raw_input('Enter a bucket name (default: {}): '
+                                  .format(default_bucket)).strip()
+          if not bucket_name:
+              bucket_name = default_bucket
+
+          if not bucket_name.startswith('gs://'):
+              bucket_name = 'gs://{}'.format(bucket_name)
+
+          p = Popen(['gsutil', 'mb', bucket_name], stdout=PIPE, stderr=PIPE)
+          out, err = p.communicate()
+          if p.returncode:
+              print err
+              if self._force_use_bucket(bucket_name, err):
+                  break
+
+        return bucket_name
+
+    def _force_use_bucket(self, bucket_name, mb_error_text):
```

Comment:
because this is returning boolean, nice for the function name to have boolean naming.
_should_force_use_bucket()